src/gax/src/http_client/mod.rs

```diff
@@ -54,18 +66,103 @@ impl ReqwestClient {
         options: crate::options::RequestOptions,
     ) -> Result<O> {
         builder = builder.bearer_auth(Self::fetch_token(&self.cred).await?);
-        if let Some(user_agent) = options.user_agent_prefix() {
+        if let Some(user_agent) = options.user_agent() {
             builder = builder.header(
                 reqwest::header::USER_AGENT,
                 reqwest::header::HeaderValue::from_str(user_agent).map_err(Error::other)?,
             );
         }
-        if let Some(timeout) = options.attempt_timeout() {
-            builder = builder.timeout(*timeout);
-        }
         if let Some(body) = body {
             builder = builder.json(&body);
         }
+        match self.make_retry_policy(&options) {
+            None => self.request_attempt::<O>(builder, &options, None).await,
+            Some(policy) => self.retry_loop::<O>(builder, &options, policy).await,
+        }
+    }
+
+    async fn retry_loop<O: serde::de::DeserializeOwned>(
+        &self,
+        builder: reqwest::RequestBuilder,
+        options: &crate::options::RequestOptions,
+        retry_policy: Box<dyn crate::retry_policy::RetryPolicy>,
+    ) -> Result<O> {
+        let throttler = self.find_retry_throttler(options);
+        let mut backoff = self.make_backoff_policy(options);
+        let mut first_attempt = true;
+        loop {
+            let builder = builder
+                .try_clone()
+                .ok_or_else(|| Error::other("cannot clone builder in retry loop".to_string()))?;
+            let remaining_time = retry_policy.remaining_time();
+            let accept = if first_attempt {
+                true
+            } else {
+                let t = throttler.lock().expect("retry throttler lock is poisoned");
+                t.throttle_retry_attempt()
+            };
+            let response = if accept {
+                self.request_attempt(builder, options, remaining_time).await
+            } else {
+                Err(Error::other("TODO change to a better error".to_string()))
+            };
+            match response {
+                Ok(r) => {
+                    if !first_attempt {
+                        throttler
+                            .lock()
+                            .expect("retry throttler lock is poisoned")
+                            .on_success();
+                    }
+                    return Ok(r);
+                }
+                Err(e) => {
+                    let delay = backoff.on_failure();
```

Comment:
I do not think the integration tests validate that the backoff policy is called.